src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -571,16 +573,26 @@ DeferErrorIfCannotPushdownSubquery(Query *subqueryTree, bool outerQueryHasLimit)
 		errorDetail = "Offset clause is currently unsupported";
 	}
 
-	if (subqueryTree->limitCount && !outerQueryHasLimit)
+	/*
+	 * Limit is partially supported when SubqueryPushdown is set.
+	 * The outermost query must have a limit clause.
+	 */
+	if (subqueryTree->limitCount && !SubqueryPushdown)
```

Comment:
> Also, in the master branch, we error out LIMITs whenever there is no LIMIT on the outside query. But, here we skip that if the flag is set?

I did not understand this comment.  here we check
1 -  inner query has limit and subquery_pushdown is not set ---> error
2 - inner query has limit, pushdown_flag is set (implicit due to else if), and outermost query does not have limit ---> error